### PR TITLE
MOO-242: Add suite options

### PR DESCRIPTION
**Pull request by tracyatrf**
adding new option --suites or -s, this will allow you to specify which suites to run or exclude.
For example:
Run just panda and champ:
`be moose staging -s champ,panda`
Run everything but unsecured:
`be moose staging --suites ~unsecured`

you get the idea.

### CODE REVIEW

File: lib/moose/core/configuration_options.rb [hunk 1]
```diff
@@ -143,6 +143,23 @@ def option_parser
             end
           end
 
+          parser.on("--suites=VALUES", Array, 'Which suites to run or exclude',
+                    "by adding ~ before the tag.",
+                    "  - e.g. ~unsecured") do |suites|
+            suites.each do |suite|
+              if suite == "all"
+                add_suite_filter(:exclusion_filters, nil)
```
Comment (kiisu5) [inline lib/moose/core/configuration_options.rb]: what if exclusion got its default value from a config file?  Then we could set it up to exclude unsecured unless specifically included without hard coding it into moose or excluding it purposefully every time.

Comment (vinchbr) [inline lib/moose/core/configuration_options.rb]: No more config files, `Moose.configuration`

Comment (tracyatrf) [inline lib/moose/core/configuration_options.rb]: I have a mutagenesis branch that sets the default filter for unsecured to be false in the Moose.configuration block.

Comment (vinchbr) [inline lib/moose/core/configuration_options.rb]: <3

Comment (kiisu5) [inline lib/moose/core/configuration_options.rb]: yay!  💃

Comment (tracyatrf) [inline lib/moose/core/configuration_options.rb]: That is why I included the "all" option here -- there needed to be a way to explicitly override and run all suites, even if I add a default value of excluding the unsecured suite.

File: lib/moose/core/configuration_options.rb [hunk 2]
```diff
@@ -143,6 +143,23 @@ def option_parser
             end
           end
 
+          parser.on("--suites=VALUES", Array, 'Which suites to run or exclude',
+                    "by adding ~ before the tag.",
+                    "  - e.g. ~unsecured") do |suites|
+            suites.each do |suite|
```
Comment (tracyatrf) [inline lib/moose/core/configuration_options.rb]: yeah mostly following form here. We are adding these filter types here, and processing them at a later time in execution. See  `lib/moose/suite/runner.rb` changes.

**Comment by kiisu5** [comment]
we also need to update the documentation in confluence to include the new moose features

File: lib/moose/core/configuration_options.rb (hunk 2 above)
Comment (rchupick) [inline lib/moose/core/configuration_options.rb]: ah i see 👍

**Pull request merged**